  INFO root [0x000036fc] [2020-10-24 23:00:38,183] - 
  INFO root [0x000036fc] [2020-10-24 23:00:38,184] - -----------------------------------------------------------------------------------------


  INFO root [0x000036fc] [2020-10-24 23:00:38,185] - Resmgr::initialize: KBE_ROOT=D:/kbengine_24pic/
  INFO root [0x000036fc] [2020-10-24 23:00:38,186] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine_24pic/kbe/res/;D:/kbengine_24pic/kbengine_stone_assets/;D:/kbengine_24pic/kbengine_stone_assets/scripts/;D:/kbengine_24pic/kbengine_stone_assets/res/
  INFO root [0x000036fc] [2020-10-24 23:00:38,186] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine_24pic/kbe/bin/server/
  INFO root [0x000036fc] [2020-10-24 23:00:38,191] - NetworkInterface::initialize(EXTERNAL): bound to all interfaces with default route interface on 192.168.2.246 ( 192.168.2.246:20015 )
  INFO root [0x000036fc] [2020-10-24 23:00:38,192] - NetworkInterface::initialize(EXTERNAL): address 192.168.2.246:20015, SOMAXCONN=128.
  INFO root [0x000036fc] [2020-10-24 23:00:38,196] - NetworkInterface::initialize(INTERNAL): bound to all interfaces with default route interface on 192.168.2.246 ( 192.168.2.246:51282 )
  INFO root [0x000036fc] [2020-10-24 23:00:38,197] - NetworkInterface::initialize(INTERNAL): address 192.168.2.246:51282, SOMAXCONN=128.
  INFO root [0x000036fc] [2020-10-24 23:00:38,197] - server-configs:
  INFO root [0x000036fc] [2020-10-24 23:00:38,197] - 	gameUpdateHertz : 30
  INFO root [0x000036fc] [2020-10-24 23:00:38,197] - 	entryScriptFile : kbemain
  INFO root [0x000036fc] [2020-10-24 23:00:38,197] - 	internalAddr : 192.168.2.246:51282
  INFO root [0x000036fc] [2020-10-24 23:00:38,198] - 	externalAddr : 192.168.2.246:20015
  INFO root [0x000036fc] [2020-10-24 23:00:38,198] - 	componentID : 720012
  INFO root [0x000036fc] [2020-10-24 23:00:39,876] - ---- baseapp Version: 1.2.7. ScriptVersion: 0.1.0. Protocol: 84173EFF21DE506F3117154D4C51407C. Config: Debug X86. Built: 19:02:14 Aug  4 2020. AppID: 720012. UID: 1937. PID: 6476 ----
  INFO root [0x000036fc] [2020-10-24 23:00:39,877] - Server : Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz with 2097151kB RAM
  INFO root [0x000036fc] [2020-10-24 23:00:39,878] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [0x000036fc] [2020-10-24 23:00:39,879] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,016] - Script::install_py_dlls(): PyInit_pyexpat
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,017] - Script::install_py_dlls(): PyInit__socket
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,017] - Script::install_py_dlls(): PyInit__ssl
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,021] - Script::install_py_dlls(): PyInit__hashlib
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,022] - Script::install_py_dlls(): PyInit_select
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,023] - Script::install_py_dlls(): PyInit__ctypes
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,024] - Script::install_py_dlls(): PyInit__elementtree
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,079] - Script::install_py_dlls(): PyInit_unicodedata
  INFO root [0x000036fc] [2020-10-24 23:00:40,119] - Script::install(): is successfully, Python=(3.4.2 (default, Aug  4 2020, 19:01:54) [MSC v.1925 32 bit (Intel)])!
  INFO root [0x000036fc] [2020-10-24 23:00:40,120] - Script::install(): KBExtra is successfully!
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,186] - loaded script:Avatar(1).
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,257] - loaded script:Halls(2).
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,258] - loaded script:Room(3).
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,260] - loaded script:PrivateRoom(4).
  INFO root [0x000036fc] [2020-10-24 23:00:40,260] - Baseapp::initializeEnd: started resmgr tick(60s)!
  INFO root [0x000036fc] [2020-10-24 23:00:40,261] - TelnetServer server is running on port 40000
  INFO root [0x000036fc] [2020-10-24 23:00:40,497] - ---- baseapp is running ----
 DEBUG root [0x000036fc] [2020-10-24 23:00:40,497] - Components::process(): Request for the process of identity...
  WARN root [0x000036fc] [2020-10-24 23:00:46,673] - EntityApp::checkTickPeriod: tick took 18712% (18.71 seconds)!
  INFO root [0x000036fc] [2020-10-24 23:00:47,698] - Components::findComponents: find logger(1)...
  INFO root [0x000036fc] [2020-10-24 23:00:47,707] - Components::findComponents: found logger, addr:192.168.2.246:51278
  INFO root [0x000036fc] [2020-10-24 23:00:47,707] - Components::addComponent: reset orderLog, uid=1937!
  INFO root [0x000036fc] [2020-10-24 23:00:47,707] - Components::addComponent[logger], uid=1937, componentID=210002, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [0x000036fc] [2020-10-24 23:00:47,708] - Components::findComponents: find dbmgr(2)...
  INFO root [0x000036fc] [2020-10-24 23:00:47,819] - Components::findComponents: found logger, addr:192.168.2.246:51278
  INFO root [0x000036fc] [2020-10-24 23:00:47,825] - Components::findComponents: found dbmgr, addr:192.168.2.246:51279
  INFO root [0x000036fc] [2020-10-24 23:00:47,825] - Components::addComponent: reset orderLog, uid=1937!
  INFO root [0x000036fc] [2020-10-24 23:00:47,826] - Components::addComponent[dbmgr], uid=1937, componentID=42002, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [0x000036fc] [2020-10-24 23:00:47,826] - Components::findComponents: found dbmgr, addr:192.168.2.246:51279
  WARN root [0x000036fc] [2020-10-24 23:00:48,826] - EntityApp::checkTickPeriod: tick took 3521% (3.52 seconds)!
  WARN root [0x000036fc] [2020-10-24 23:00:48,827] - Forwarding logs to logger[192.168.2.246:51278/0/0/0]...
 ERROR root [0x000036fc] [2020-10-24 23:57:22,960] - Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [0x000036fc] [2020-10-24 23:57:22,960] - The following logs sent to logger failed:
  WARN root [0x000036fc] [2020-10-24 23:57:22,960] - ==> [2020-10-24 23:57:22 966] TCPPacketReceiver::processPendingEvents(192.168.2.246:51278): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [0x000036fc] [2020-10-24 23:57:22,960] - ==> [2020-10-24 23:57:22 967] Components::removeComponentByChannel: logger : 210002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  WARN root [0x000036fc] [2020-10-24 23:57:23,038] - TCPPacketReceiver::processPendingEvents(192.168.2.246:51279): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [0x000036fc] [2020-10-24 23:57:23,038] - Components::removeComponentByChannel: dbmgr : 42002, Abnormal exit! TCPPacketReceiver::processRecv(): errno=10054

  INFO root [0x000036fc] [2020-10-24 23:57:23,038] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [0x000036fc] [2020-10-24 23:57:23,038] - onBaseAppShutDown: state=0
  INFO root [0x000036fc] [2020-10-24 23:57:23,038] - Shutdowner::onShutdown: shutting down
S_INFO root [0x000036fc] [2020-10-24 23:57:23,038] - onBaseAppShutDown: state=1
  INFO root [0x000036fc] [2020-10-24 23:57:23,038] - Shutdowner::onShutdownEnd: shutting down
S_INFO root [0x000036fc] [2020-10-24 23:57:23,038] - onBaseAppShutDown: state=2
 ERROR root [0x000036fc] [2020-10-24 23:57:23,070] - Avatar::onCellWriteToDBCompleted(10008): not found dbmgr!
 ERROR root [0x000036fc] [2020-10-24 23:57:23,070] - Entity::onDestroy: not found dbmgr!
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,070] - Avatar::~Avatar(): 10008
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,070] - Avatar::~Avatar(): 10007
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,070] - PrivateRoom::~PrivateRoom(): 10006
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,070] - Halls::~Halls(): 10001
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,085] - Script::uninstall_py_dlls(): PyInit_pyexpat
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,085] - Script::uninstall_py_dlls(): PyInit__socket
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,085] - Script::uninstall_py_dlls(): PyInit__ssl
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,085] - Script::uninstall_py_dlls(): PyInit__hashlib
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,085] - Script::uninstall_py_dlls(): PyInit_select
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,085] - Script::uninstall_py_dlls(): PyInit__ctypes
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,085] - Script::uninstall_py_dlls(): PyInit__elementtree
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,085] - Script::uninstall_py_dlls(): PyInit_unicodedata
  INFO root [0x000036fc] [2020-10-24 23:57:23,101] - Script::uninstall(): is successfully!
 DEBUG root [0x000036fc] [2020-10-24 23:57:23,101] - ThreadPool::destroy(): starting size 2.
